layout/save.go

```diff
@@ -27,6 +28,51 @@ func (i *Image) SaveAs(name string, additionalNames ...string) error {
 		ops = append(ops, WithoutLayers())
 	}
 
+	i.Image, err = imgutil.MutateManifest(i, func(mfest *v1.Manifest) {
+		config := mfest.Config
+		if annos, _ := i.Annotations(); len(annos) != 0 {
+			mfest.Annotations = annos
+			config.Annotations = annos
+		}
+
+		if urls, _ := i.URLs(); len(urls) != 0 {
+			config.URLs = append(config.URLs, urls...)
+		}
+
+		if config.Platform == nil {
+			config.Platform = &v1.Platform{}
+		}
+
+		if features, _ := i.Features(); len(features) != 0 {
+			config.Platform.Features = append(config.Platform.Features, features...)
+		}
+
+		if osFeatures, _ := i.OSFeatures(); len(osFeatures) != 0 {
+			config.Platform.OSFeatures = append(config.Platform.OSFeatures, osFeatures...)
+		}
+
+		if os, _ := i.OS(); os != "" {
+			config.Platform.OS = os
+		}
+
+		if arch, _ := i.Architecture(); arch != "" {
+			config.Platform.Architecture = arch
+		}
+
+		if variant, _ := i.Variant(); variant != "" {
+			config.Platform.Variant = variant
+		}
+
+		if osVersion, _ := i.OSVersion(); osVersion != "" {
+			config.Platform.OSVersion = osVersion
+		}
+
+		mfest.Config = config
+	})
+	if err != nil {
+		return err
+	}
+
```

Comment:
@WYGIN actually these lines are causing the timeout when the Layout tests are executed, if I remove it, current tests are fine

Reply:
The issue is raised due to concurrency, I just noticed it and fixed it.